api/src/main/java/org/openmrs/module/legacyui/api/impl/LegacyUIImpl.java

```diff
@@ -0,0 +1,291 @@
+		
+		// need to discontinue any open orders for this patient
+		discontinueAllDrugOrders(patient, reasonForExit, dateExited);
+	}
+	
+	/**
+	 * Copied from OpenMRS core 1.12.x 
+	 * See https://github.com/openmrs/openmrs-core/blob/1.12.x/api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java#L1219
+	 * 
+	 * TODO: Patients should actually be allowed to exit multiple times
+	 * 
+	 * @param patient
+	 * @param exitDate
+	 * @param cause
+	 */
+	private void saveReasonForExitObs(Patient patient, Date exitDate, Concept cause) throws APIException {
+		
+		if (patient == null) {
+			throw new APIException("Patient supplied to method is null");
+		}
+		if (exitDate == null) {
+			throw new APIException("Exit date supplied to method is null");
+		}
+		if (cause == null) {
+			throw new APIException("Cause supplied to method is null");
+		}
+		
+		// need to make sure there is an Obs that represents the patient's
+		// exit
+		log.debug("Patient is exiting, so let's make sure there's an Obs for it");
+		
+		String codProp = Context.getAdministrationService().getGlobalProperty("concept.reasonExitedCare");
+		Concept reasonForExit = Context.getConceptService().getConcept(codProp);
+		
+		if (reasonForExit != null) {
+			List<Obs> obssExit = Context.getObsService().getObservationsByPersonAndConcept(patient, reasonForExit);
+			if (obssExit != null) {
+				if (obssExit.size() > 1) {
+					log.error("Multiple reasons for exit (" + obssExit.size() + ")?  Shouldn't be...");
+				} else {
+					Obs obsExit = null;
+					if (obssExit.size() == 1) {
+						// already has a reason for exit - let's edit it.
+						log.debug("Already has a reason for exit, so changing it");
+						
+						obsExit = obssExit.iterator().next();
+						
+					} else {
+						// no reason for exit obs yet, so let's make one
+						log.debug("No reason for exit yet, let's create one.");
+						
+						obsExit = new Obs();
+						obsExit.setPerson(patient);
+						obsExit.setConcept(reasonForExit);
+						
+						Location loc = Context.getLocationService().getDefaultLocation();
+						
+						if (loc != null) {
+							obsExit.setLocation(loc);
+						} else {
+							log.error("Could not find a suitable location for which to create this new Obs");
+						}
+					}
+					
+					if (obsExit != null) {
+						// put the right concept and (maybe) text in this
+						// obs
+						obsExit.setValueCoded(cause);
+						obsExit.setValueCodedName(cause.getName()); // ABKTODO: presume current locale?
+						obsExit.setObsDatetime(exitDate);
+						Context.getObsService().saveObs(obsExit, "updated by PatientService.saveReasonForExit");
+					}
+				}
+			}
+		} else {
+			log.debug("Reason for exit is null - should not have gotten here without throwing an error on the form.");
+		}
+		
+	}
+	
+	/**
+	 * Copied from OpenMRS core 1.12.x
+	 * See https://github.com/openmrs/openmrs-core/blob/1.12.x/api/src/main/java/org/openmrs/api/impl/ProgramWorkflowServiceImpl.java#L450
```

Comment:
Same as above for using tags

Reply:
[Fixed](https://github.com/openmrs/openmrs-module-legacyui/pull/128/commits/9dd92c8a5560377daedb2e231c639e315fa177b8)